src-electron/generator/helper-attribute.js

```diff
@@ -27,6 +27,40 @@ const templateUtil = require('./template-util')
 const zclUtil = require('../util/zcl-util')
 const dbEnum = require('../../src-shared/db-enum')
 
+/**
+ * Counts the number of mandatory attributes (not optional, non-global) for the current cluster context.
+ * Usage: {{count_mandatory_attributes side="server"}}
+ * If side is not provided or invalid, counts all attributes.
+ */
+async function count_mandatory_attributes(options) {
+  if (!('id' in this))
+    throw new Error(
+      'count_mandatory_attributes requires an id inside the context.'
```

Comment:
Also add that this should be used under a cluster based block helper. The id has to be a reference to a cluster id?